PurchasesCoreSwiftTests/SubscriberAttributes/AttributionDataMigratorTests.swift

```diff
@@ -0,0 +1,732 @@
+                defaultValue: AttributionDataMigratorTests.defaultIdfa)
+        function(keyPresence: idfv, data: &data, key: AttributionKey.idfv.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIdfv)
+        function(keyPresence: ip, data: &data, key: AttributionKey.ip.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIp)
+        function(keyPresence: channel, data: &data, key: AttributionKey.Branch.channel.rawValue,
+                defaultValue: "Facebook")
+        function(keyPresence: campaign, data: &data, key: AttributionKey.Branch.campaign.rawValue,
+                defaultValue: "Facebook Ads 01293")
+
+        return data
+    }
+
+    func mParticleData(withIDFA idfa: KeyPresence = .defaultValue,
+                       idfv: KeyPresence = .defaultValue,
+                       mParticleId: KeyPresence = .defaultValue,
+                       networkID: KeyPresence = .notPresent,
+                       ip: KeyPresence = .defaultValue) -> [String: Any] {
+        var data: [String: Any] = [:]
+
+        function(keyPresence: idfa, data: &data, key: AttributionKey.idfa.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIdfa)
+        function(keyPresence: idfv, data: &data, key: AttributionKey.idfv.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIdfv)
+        function(keyPresence: mParticleId, data: &data, key: AttributionKey.MParticle.id.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultNetworkId)
+        function(keyPresence: networkID, data: &data, key: AttributionKey.networkID.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultRCNetworkId)
+        function(keyPresence: ip, data: &data, key: AttributionKey.ip.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIp)
+        return data
+    }
+
+    func facebookOrTenjinData(withIDFA idfa: KeyPresence = .defaultValue,
+                              idfv: KeyPresence = .defaultValue, ip: KeyPresence = .defaultValue) -> [String: Any] {
+        var data: [String: Any] = [:]
+        function(keyPresence: idfa, data: &data, key: AttributionKey.idfa.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIdfa)
+        function(keyPresence: idfv, data: &data, key: AttributionKey.idfv.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIdfv)
+        function(keyPresence: ip, data: &data, key: AttributionKey.ip.rawValue,
+                defaultValue: AttributionDataMigratorTests.defaultIp)
+        return data
+    }
+
+    private func function(keyPresence: KeyPresence, data: inout [String: Any], key: String, defaultValue: String) {
+        switch keyPresence {
+        case .defaultValue:
+            data[key] = defaultValue
+        case .nsNull:
+            data[key] = NSNull()
+        case .notPresent:
+            break
+        }
```

Comment:
perhaps we can improve the naming for this `func`? `updateMappingInData?`
Also, super nitpicky but I'd usually have the `inout` param as the first item, so it's clear that _that's_ the item being modified. 
It'd also allow for naming like: `updateMapping(inData: data, keyPresence: .nsNull, key: "idfa", defaultValue: "idfa")`

Reply:
oh yes I forgot to update hahaha `function` lol 😆